pandas/core/generic.py

```diff
@@ -4868,32 +4869,32 @@ def abs(self):
     @Appender(_shared_docs['describe'] % _shared_doc_kwargs)
     def describe(self, percentiles=None, include=None, exclude=None):
         if self.ndim >= 3:
-            msg = "describe is not implemented on on Panel or PanelND objects."
+            msg = "describe is not implemented on Panel or PanelND objects."
             raise NotImplementedError(msg)
+        elif self.ndim == 2 and self.columns.size == 0:
+            raise ValueError("Cannot describe a DataFrame without columns")
```

Comment:
No, it's a non-breaking change. `pd.concat()` inside `describe()`raises anyway (I put it in one of my earlier comments). The purpose of this one is just to give a more meaningful message (and skip some code).